processor/samplingprocessor/processor.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package samplingprocessor
+
+import (
+	"context"
+	"math/rand"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type throughputMeasurementProcessor struct {
+	logger          *zap.Logger
+	dropCutOffRatio float64
+}
+
+func newThroughputMeasurementProcessor(logger *zap.Logger, cfg *Config) *throughputMeasurementProcessor {
+	return &throughputMeasurementProcessor{
+		logger:          logger,
+		dropCutOffRatio: cfg.DropRatio,
+	}
+}
+
+func (tmp *throughputMeasurementProcessor) processTraces(_ context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+	//#nosec G404 -- randomly generated number is not used for security purposes. It's ok if it's weak
+	if rand.Float64() <= tmp.dropCutOffRatio {
+		return ptrace.NewTraces(), nil
+	}
+
+	return td, nil
+}
```

Comment:
Note here, there is https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/probabilisticsamplerprocessor that does this for traces in contrib, but I don't really know what the differences are. The "probabilisticsampler" processor seems to have some extra considerations for distributed tracing stuff, so I'm not sure if that's 100% needed or not.

Reply:
Yeah we don't need anything super fancy for this first pass. Mostly as long as it does rough sampling we can always tweak it down the road.